File: c/claude/p13_nre.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>

static const int id_sequence[] = {5, 15, 25, 35, 45, 55, 65, 75, 85, 95};
static const int ID_SEQ_LEN = (int)(sizeof(id_sequence) / sizeof(id_sequence[0]));

/*
 * getValueFromArray: takes only an index; the array is the module-level
 * id_sequence constant.
 *
 * Returns id_sequence[index] if 0 <= index < ID_SEQ_LEN, else -1.
 */
int getValueFromArray(int index)
{
    if (index < 0 || index >= ID_SEQ_LEN) return -1;
    return id_sequence[index];
}

/*
 * try_parse: wraps strtol with full validation.
 * Returns 1 and sets *out on success; 0 on any failure.
 */
static int try_parse(const char *str, long *out)
{
    if (str == NULL || *str == '\0') return 0;
    char *end;
    errno = 0;
    long v = strtol(str, &end, 10);
    if (errno != 0 || end == str || *end != '\0') return 0;
    *out = v;
    return 1;
}

int main(void)
{
    char buf[64];

    printf("id_sequence has %d elements (indices 0–%d).\n",
           ID_SEQ_LEN, ID_SEQ_LEN - 1);
    printf("Enter an index: ");
    fflush(stdout);

    /* read input as a string */
    if (fgets(buf, sizeof(buf), stdin) == NULL) {
        fprintf(stderr, "Error: failed to read input.\n");
        return -1;
    }
    buf[strcspn(buf, "\n")] = '\0';

    long index;
    /* if conversion fails → return -1 */
    if (!try_parse(buf, &index)) {
        fprintf(stderr, "Error: \"%s\" is not a valid integer.\n", buf);
        return -1;
    }

    /* if index < 0 or index >= array_length → return -1; otherwise return id_sequence[index] */
    int result = getValueFromArray((int)index);
    if (result == -1) {
        fprintf(stderr,
                "Error: index %ld is out of bounds (valid range: 0–%d).\n",
                index, ID_SEQ_LEN - 1);
    } else {
        printf("id_sequence[%ld] = %d\n", index, result);
    }
    return result;
}

File: java/claude/p13_nre.java
import java.util.Scanner;

public class p13_nre {

    private static final int[] ID_SEQUENCE = {5, 15, 25, 35, 45, 55, 65, 75, 85, 95};

    /**
     * Returns {@code ID_SEQUENCE[index]} if {@code 0 <= index < ID_SEQUENCE.length},
     * otherwise -1. The array is the class-level constant; only an index is required.
     */
    public static int getValueFromArray(int index) {
        if (index < 0 || index >= ID_SEQUENCE.length) return -1;
        return ID_SEQUENCE[index];
    }

    public static void main(String[] args) {
        System.out.println("ID_SEQUENCE has " + ID_SEQUENCE.length
                + " elements (indices 0–" + (ID_SEQUENCE.length - 1) + ").");
        System.out.print("Enter an index: ");
        System.out.flush();

        Scanner sc = new Scanner(System.in);

        /* read input as a string */
        if (!sc.hasNextLine()) {
            System.err.println("Error: no input received.");
            sc.close();
            System.exit(-1);
        }
        String input = sc.nextLine().trim();
        sc.close();

        int index;
        try {
            /* if conversion fails → return -1 */
            index = Integer.parseInt(input);
        } catch (NumberFormatException e) {
            System.err.println("Error: \"" + input + "\" is not a valid integer.");
            System.exit(-1);
            return;
        }

        /* if index < 0 or index >= array_length → return -1; otherwise return id_sequence[index] */
        int result = getValueFromArray(index);
        if (result == -1) {
            System.err.println("Error: index " + index
                    + " is out of bounds (valid range: 0–" + (ID_SEQUENCE.length - 1) + ").");
        } else {
            System.out.println("ID_SEQUENCE[" + index + "] = " + result);
        }
        System.exit(result);
    }
}
